android-components/components/feature/search/src/main/java/mozilla/components/feature/search/storage/SearchEngineReader.kt

```diff
@@ -47,7 +64,11 @@ internal class SearchEngineReader(
             type = type,
             resultUrls = resultsUrls,
             suggestUrl = suggestUrl,
+            isGeneral = type == SearchEngine.Type.CUSTOM || isGeneralSearchEngine(identifier),
```

Comment:
I don't know if I am happy with this check all custom search engines to be isGeneral. I could also do something similar to above with the variables `name`, `suggestUrl`, etc which will be set when it's reading the file instead of deriving `isGeneral` from the type.